### DISCUSSION TITLE
win.make_modal(true) will always be displayed in front of all software in mac

### DISCUSSION THREAD

**Discussion by coolH-alt** [fltk-rs/fltk-rs#1444] [Q&A]
Now I have a component that needs to use ```window::Window```,  I need it to be displayed directly in front of my software。
use ```make_modal(true)```，It can also cover my software on Mac, but it will also cover other software。
On Win10, it only covers my software。

**Comment by MoAlyousef** [comment]
Making a window topmost is a platform feature which would require calling into platform specific code. You can try for example:
```rust
fn make_topmost(w: &window::Window) {
    assert!(w.shown());
    use std::os::raw::*;
    const SWP_NOSIZE: u32 = 0x0001; // ignore the width and height params
    const SWP_NOMOVE: u32 = 0x0002; // ignore the x and y params
    pub type HWND = *mut c_void;
    extern "C" {
        pub fn SetWindowPos(
            hwnd: HWND,
            hwnd: HWND,
            x: i32,
            y: i32,
            cx: i32,
            cy: i32,
            uflags: u32,
        ) -> i32;
    }
    unsafe {
        let handle = w.raw_handle();
        let val: *mut i64 = std::mem::transmute(-1 as i64);
        SetWindowPos(handle as _, val as _, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE);
    }
}
```

**Reply by coolH-alt** [reply]
```
let app = app::App::default().with_scheme(app::AppScheme::Gtk);
let mut win = Window::default()
    .with_size(800, 600)
    .center_screen()
    .with_label("Win1");
let mut html_win = Window::new(0, 0, 100, 100, None).center_of_parent();
html_win.set_color(Color::Black);
html_win.end();
win.end();
win.show();
fltk_webview::Webview::create(false, &mut html_win);
let mut win2 = Window::new(10, 10, 600, 300, "Win2").center_screen();
let mut btn = Button::new(100, 100, 100, 32, "Test");
btn.set_callback(move |_| println!("123"));
win2.make_modal(true);
win2.set_color(Color::Red);
win2.end();
win2.show();
// fltk_webview::Webview::create(false, &mut html_win);
app.run().unwrap();
```
Like this, if I use ```fltk_webview::Webview::create(false, &mut html_win)``` before win2 is created, on mac, win2 will be placed on top of all software.

Use ```fltk_webview::Webview::create(false, &mut html_win)``` after ```win2.show()```, the button in win2 cannot be clicked

use ```fltk_webview::Webview::create(false, &mut html_win)``` before win2 is created
![51bdacd791f0cc4abc4daa412cb54d4](https://github.com/fltk-rs/fltk-rs/assets/73577805/66ddb1dd-afcd-4d45-a0d6-08a9710d307d)

Use ```fltk_webview::Webview::create(false, &mut html_win)``` after ```win2.show()```
![68bdc5da1c0f795c91053e4841b8b82](https://github.com/fltk-rs/fltk-rs/assets/73577805/fbd8c982-e329-47a3-8ae7-9aedaf8fe67c)

**Reply by MoAlyousef** [reply]
I’ll have to investigate. It seems to be an issue of fltk-webview.

**Reply by coolH-alt** [reply]
one more question
```
let app = app::App::default().with_scheme(app::AppScheme::Gtk);
let mut win = Window::default()
    .with_size(800, 600)
    .center_screen()
    .with_label("Win1");
let mut html_win = Window::new(0, 0, 100, 100, None).center_of_parent();
html_win.set_color(Color::Black);
html_win.end();
win.end();
win.show();
let mut win2 = Window::new(10, 10, 600, 300, "Win2").center_screen();
let mut btn = Button::new(100, 100, 100, 32, "Test");
btn.set_callback(move |_| println!("123"));
win2.set_border(false);
win2.make_modal(true);
win2.set_color(Color::Red);
win2.end();
win2.show();
let mut win3 = Window::new(10, 10, 300, 300, "WIN3").center_screen();
win3.set_border(false);
win3.make_modal(true);
win3.end();
win3.show();
app.run().unwrap();
```
On mac, like this, clicking win2 win2 will cover win3, because win3 is smaller than win2, so my page will be stuck, is there a way to click win2, but win3 will not be covered by win2。
It is normal to write like this on win10. Clicking on win2 will not overwrite win3

**Reply by MoAlyousef** [reply]
I think it's the fact the webview window calls cocoa's makeKeyAndOrderFront which appears to interfere with fltk's make_modal call (in ordering at least). I'm not sure there's a way around it when using fltk-webview.

**Reply by coolH-alt** [reply]
Please help me check the above questions. If not, I will change the writing method

**Reply by MoAlyousef** [reply]
Can you try win3.set_on_top()
